components/camera/platforms/jetson/camera.go

```diff
@@ -0,0 +1,114 @@
+package jetsoncamera
+
+// #include <sys/utsname.h>
+// #include <string.h>
+import "C"
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"runtime"
+	"unsafe"
+)
+
+// GetOSInformation pulls relevant OS attributes as an OSInformation struct
+// Kernel and Device will be "unkown" if unable to retrieve info from the filesystem
+// returns an error if kernel version or device name is unavailable
+func DetectOSInformation() (OSInformation, error) {
+	kernelVersion, err := getKernelVersion()
+	if err != nil {
+		return OSInformation{}, fmt.Errorf("failed to get kernel version: %w", err)
+	}
+	deviceName, err := getDeviceName()
+	if err != nil {
+		return OSInformation{}, fmt.Errorf("failed to get device name: %w", err)
+	}
+	osInfo := OSInformation{
+		Name:   runtime.GOOS,
+		Arch:   runtime.GOARCH,
+		Kernel: kernelVersion,
+		Device: deviceName,
+	}
+	return osInfo, nil
+}
+
+// getKernelVersion returns the Linux kernel version ($ uname -r)
+func getKernelVersion() (string, error) {
+	var utsName C.struct_utsname
+	if C.uname(&utsName) == -1 {
+		return Unknown, fmt.Errorf("uname information unavailable (%v)", utsName)
+	}
+	release := C.GoString((*C.char)(unsafe.Pointer(&utsName.release[0])))
+	return release, nil
+}
+
+// getDeviceName returns the model name of the device
+// ($ cat /sys/firmware/devicetree/base/model)
+func getDeviceName() (string, error) {
+	devicePath := "/sys/firmware/devicetree/base/model"
```

Comment:
This directory doesn't exist on x86 boards like a Framework laptop or an Intel Up 4000 (x86 doesn't do any devicetree-related things). I recognize that this function is in a directory for Jetson boards, but it seems like it might be called from _any_ board given the rest of the PR. ...and maybe that's okay for the current features that use this (maybe it's okay to return "Unknown" for all x86 boards). but this is somewhat brittle because it doesn't quite do what the function name implies it should do.